src/FactoryTrait.php

```diff
@@ -33,19 +33,36 @@ public function mergeSeeds($seed, $seed2, ...$more_seeds)
             $seed2 = $this->mergeSeeds($seed2, ...$more_seeds);
         }
 
-        if (is_object($seed)) {
+        if (is_object($seed) || is_object($seed2)) {
+            if (is_object($seed)) {
+                $passively = true; // set defaults but don't override existing properties
+            } else {
+                $passively = false;
+                [$seed, $seed2] = [$seed2, $seed]; // swap seeds
```

Comment:
This is clever line. I always used $tmp variable for that, but this way it's much cleaner.